### DISCUSSION TITLE
HELP NEEDED: Any way to recover data on Synology NAS?

### DISCUSSION THREAD

**Discussion by goaliedude3919** [TriliumNext/Trilium#3688] [Q&A]
I have Trilium installed on my Synology NAS via Docker (followed [these instructions](https://mariushosting.com/how-to-install-trilium-on-your-synology-nas/))

I realized today that, despite having "-v /volume1/docker/trilium:/home/node/trilium-data" there was nothing in my /volume1/docker/trilium folder. I thought the solution to this would be to manually map the volume [like so](https://i.imgur.com/oCGxQyp.png). However after doing that and restarting, it brought me to the initial setup screen. I tried to go through the option of saying I had a server already that I wanted to sync, but that did not work. I immediately stop the container, removed the volume mapping, and restarted. Unfortunately, it still brings me to the initial setup screen.

I'm worried that I've now lost everything. I assumed that mapping the volume would put the existing data into /volume1/docker/trilium, but it seems to have worked in the opposite direction and instead wiped out /home/node/trilium-data. Is there any way to recover my data?

**Comment by dumblegore** [comment]
I installed a second instance with mapping and synced it with the first.

**Comment by zadam** [comment]
If the data is stored within the container, then it should be recoverable (e.g. using the sync as @dumblegore suggested, or perhaps exporting all content in ZIP). Just make sure to not delete the container.

**Comment by goaliedude3919** [comment]
I tried setting up another container and doing a sync to the original one (which I did make sure is running, but I get this error:
`Sync setup failed: Request to get http://<ip_address>:<port>/api/setup/sync-seed failed, error: 400 Bad Request`

I also tried using localhost instead of the IP address but that gave me this error:
`http://localhost:<port>/api/setup/sync-seed failed, error: Error: connect ECONNREFUSED 127.0.0.1:<port>`

The last thing I tried was using my external IP instead of internal but that gave me the same error essentially as using localhost.

Does that mean I'm pretty much screwed?

**Reply by dumblegore** [reply]
You have to set the port# of both instances manually.

**Reply by goaliedude3919** [reply]
Each instance has its own port and is running simultaneously. 

Oh, I think I see why you said that. The port numbers were there, but since I didn't put the text in a code block, it didn't show the  `<port>` that I had put in.
